server/protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationStrategy.java

```diff
@@ -18,13 +18,13 @@
  ****************************************************************/
 package org.apache.james.jmap;
 
-import java.util.stream.Stream;
+import javax.servlet.http.HttpServletRequest;
 
 import org.apache.james.mailbox.MailboxSession;
 
 public interface AuthenticationStrategy {
 
-    MailboxSession createMailboxSession(Stream<String> requestHeaders);
+    MailboxSession createMailboxSession(HttpServletRequest httpRequest);
```

Comment:
I don't think so : you want to create a session and it should already throw when authorization is not ok. Actually, you only have to remove `checkAuthorizationHeader` and everything else is already ok